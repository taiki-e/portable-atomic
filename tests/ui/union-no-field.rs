// This is rustc error.

use portable_atomic::Atomicable;

#[derive(Clone, Copy, Atomicable)]
union A {}

fn main() {}
