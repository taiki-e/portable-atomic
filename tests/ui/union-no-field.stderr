error: #[derive(Atomicable)] may not be used on union with no field
 --> tests/ui/union-no-field.rs:6:7
  |
6 | union A {}
  |       ^

error: unions cannot have zero fields
 --> tests/ui/union-no-field.rs:6:1
  |
6 | union A {}
  | ^^^^^^^^^^
